src/Bindings/ManualBindings.cpp

```diff
@@ -3967,6 +3967,227 @@ static int tolua_cEntity_GetSpeed(lua_State * tolua_S)
 
 
 
+static int tolua_cBookContent_AddPage(lua_State * tolua_S)
+{
+	// cBookContent:AddPage(string)
+	// cBookContent:AddPage(cCompositeChat)
+
+	cLuaState L(tolua_S);
+	if (
+		!L.CheckParamSelf("cBookContent") ||
+		lua_isnil(L, 2) ||
+		!L.CheckParamEnd(3)
+	)
+	{
+		return 0;
+	}
+
+	cBookContent * BookContent = nullptr;
+	L.GetStackValue(1, BookContent);
+
+	// Check type of second param
+	if (L.GetTypeText(2) == "string")
+	{
+		AString Page;
+		L.GetStackValue(2, Page);
+		BookContent->AddPage(Page);
+	}
+	else if (L.CheckParamUserType(2, "cCompositeChat"))
+	{
+		cCompositeChat * CompositeChat = nullptr;
+		L.GetStackValue(2, CompositeChat);
+		if (BookContent->IsSigned())
+		{
+			BookContent->AddPage(CompositeChat->CreateJsonString());
+		}
+		else
+		{
+			BookContent->AddPage(CompositeChat->ExtractText());
```

Comment:
This will remove all formatting that could have been applied to the `cCompositeChat`. We might want to keep it, using the old-style formatting codes, or is that not possible?

Reply:
Ah well this is a problem. What do you mean with old style?

Reply:
Old-style formatting uses `§` in the text to switch colors and styles. The new formatting uses JSON for the different parts.